test/fluree/db/nameservice_query_test.clj

```diff
@@ -0,0 +1,133 @@
+(ns fluree.db.nameservice-query-test
+  (:require [babashka.fs :refer [with-temp-dir]]
+            [clojure.test :refer [deftest is testing]]
+            [fluree.db.api :as fluree]
+            [fluree.db.json-ld.iri :as iri]))
+
+(deftest nameservice-query-test
+  (testing "Nameservice query functionality"
+    (let [conn @(fluree/connect-memory {})]
+      (try
+        ;; Create multiple ledgers with some data
+        (testing "Create ledgers and insert data"
+          ;; Create first ledger with some data
+          @(fluree/create conn "ledger-one" {})
+          @(fluree/insert! conn "ledger-one"
+                           {"@context" {"test" "http://example.org/test#"}
+                            "@graph" [{"@id" "test:person1"
+                                       "@type" "Person"
+                                       "name" "Alice"}]})
+
+          ;; Create second ledger with different data
+          @(fluree/create conn "ledger-two" {})
+          @(fluree/insert! conn "ledger-two"
+                           {"@context" {"test" "http://example.org/test#"}
+                            "@graph" [{"@id" "test:person2"
+                                       "@type" "Person"
+                                       "name" "Bob"}]})
+
+          ;; Create third ledger with more data (multiple commits)
+          @(fluree/create conn "ledger-three" {})
+          @(fluree/insert! conn "ledger-three"
+                           {"@context" {"test" "http://example.org/test#"}
+                            "@graph" [{"@id" "test:person3"
+                                       "@type" "Person"
+                                       "name" "Charlie"}]})
+          @(fluree/insert! conn "ledger-three"
+                           {"@context" {"test" "http://example.org/test#"}
+                            "@graph" [{"@id" "test:person4"
+                                       "@type" "Person"
+                                       "name" "David"}]}))
+
+        (testing "Query all nameservice records"
+          (let [query {"select" ["?s" "?p" "?o"]
+                       "where" [{"@id" "?s" "?p" "?o"}]}
+                result @(fluree/query-nameservice conn query {})]
+            ;; Should have data from all ledgers plus temporary ledger metadata
+            (is (> (count result) 10) "Should have multiple records")
```

Comment:
I think the annotation here should be more precised. 9 records returned still counts as "multiple" even though it would cause the test to fail